src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -97,7 +97,7 @@ export class NotebookEditor implements INotebookEditor {
                 }
             })
         );
-        disposables.push(model.onDidDispose(this._closed.fire.bind(this._closed, this)));
+        disposables.push(model.onDidDispose(this.dispose.bind(this)));
```

Comment:
This is the change that I made after checking out how this works shutting down VS. When you shut down VS you do get a dispose on the notebook editor, but you do not get a close. In the notebookEditorProvider the disposeResourceRelatedToNotebookEditor is now only called onClose, not on dispose (since dispose is happening more often, such as moving panes).

disposeResourceRelatedToNotebookEditor basically does the following:
run closeEditor in notebookEditorProvider
dispose editor in notebookEditor
dispose model in baseModel

In the shutdown case the model is disposed via the notebookStorageProvider which puts all models on the dispose queue. In the notebookEditorProvider onEditorOpened we hook up to run the closedEditor function when the editor is closed. And the notebookEditor is set to call the .closed event when the model is disposed. Meaning that on shutdown the following happens:

dispose model in baseModel
disposing model causes closed to file in notebookEditor
run closeEditor in notebookEditorProvider

So basically the exact same things happen here. However I did note that notebookEditor dispose just fires the closed event. But it only hooks up firing the closed event to when the model is disposed not the full dispose. What I have changed this to now is functionally the same, since the only thing that dispose is doing is firing the closed event. But I think this makes it a bit safer in the future if we want to add more to the dispose here.

Reply:
@DonJayamanne @rchiodo @joyceerhl Made a tweak on the review. Given that I'm planning on porting this to release wanted to triple check on signoff on it.